core/src/main/scala/org/apache/spark/storage/BlockId.scala

```diff
@@ -51,19 +51,39 @@ case class RDDBlockId(rddId: Int, splitIndex: Int) extends BlockId {
 
 // Format of the shuffle block ids (including data and index) should be kept in sync with
 // org.apache.spark.network.shuffle.ExternalShuffleBlockResolver#getBlockData().
+// IndeterminateAttemptId is only set when the ShuffleMapStage's [[DeterministicLevel]] is
+// INDETERMINATE and fetch fail triggered whole map stage rerun.
 @DeveloperApi
-case class ShuffleBlockId(shuffleId: Int, mapId: Int, reduceId: Int) extends BlockId {
-  override def name: String = "shuffle_" + shuffleId + "_" + mapId + "_" + reduceId
+case class ShuffleBlockId(
+    shuffleId: Int,
+    mapId: Int,
+    reduceId: Int,
+    stageAttemptId: Int = 0)
```

Comment:
The default parameter just for less change of existing tests, if we don't agree, I'll change all tests and delete the default value.